src/Invoice_GenUI/Invoice_GenUI/Models/Validation/ContactValidation.cs

```diff
@@ -0,0 +1,28 @@
+﻿
+
+using System.Globalization;
+using System.Text.RegularExpressions;
+using System.Windows.Controls;
+
+namespace Invoice_GenUI.Models.Validation;
+
+public class ContactValidation : ValidationRule
+{
+    public override ValidationResult Validate(object value, CultureInfo cultureInfo)
+    {
+        Regex regex = new Regex("^(?:(?:0(?:0|11)[\\s-]?|\\+)44[\\s-]?(?:0[\\s-]?)?|0)(?:\\d{2}[\\s-]?\\d{4}[\\s-]?\\d{4}|\\d{3}[\\s-]?\\d{3}[\\s-]?\\d{3,4}|\\d{4}[\\s-]?(?:\\d{5}|\\d{3}[\\s-]?\\d{3})|\\d{5}[\\s-]?\\d{4,5}|8(?:00[\\s-]?11[\\s-]?11|45[\\s-]?46[\\s-]?4\\d))$");
```

Comment:
:question: can you explain what this regex is doing, or have you copy-pasted it from the web somewhere?

Reply:
I got it from the regex library and tested it out in the editor using different, random phone numbers.

Reply:
So does this work only with UK phone numbers?

Reply:
Same answer as the postcode pattern, this pattern was made to validate UK phone numbers however it is possible to match with non-uk phone number as well, I should of mentioned it previously my bad.